### ISSUE TITLE
Active Response Still Referencing ar.conf rather than merged.conf

### ISSUE THREAD

**Issue by 32bitbradley** [wazuh/wazuh#5288]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.12.3 | Wazuh component | Manager and Agent | Packages | Range of Centos, Ubuntu and Windows |

Hi everyone, I have a weird bug here.

So after double-checking how active response works via the documentation, it's my understanding that the file `/var/ossec/etc/shared/ar.conf` file gets merged with `/var/ossec/shared/*/agent.conf` files, and then the resulting file `/var/ossec/etc/shared/merged.conf` is shipped off the the related agents in the related groups. Please do correct me if I've got this wrong :)

As a result of this process, the `/var/ossec/etc/shared/ar.conf` on the manager, never actually gets physically copied to the agent to the same file location `/var/ossec/etc/shared/ar.conf`. 

When running active response, however, I can see that this file `/var/ossec/etc/shared/ar.conf` is still referenced by the agent when checking if the active response command is valid., but as it's been merged into merged.conf instead, it doesn't exist on the agent, so active responses don't run.

```
2020/06/22 16:58:21 ossec-execd: ERROR: (1103): Could not open file '/var/ossec/etc/shared/ar.conf' due to [(2)-(No s
uch file or directory)].
```

I've noticed that the default Wazuh RPM and DEBS also contain a file called `/var/ossec/etc/shared/default/ar.conf` (In the default group) in addition to `/var/ossec/etc/shared/ar.conf`. Out of the box, these files are then same, which coincidentally results in the `/var/ossec/etc/shared/ar.conf` file existing on the agent out of the box, which means active response works out of the box.

We run a lot of custom active response scripts, so we have a different `/var/ossec/etc/shared/ar.conf` (generated by the Wazuh Manager as it should), and this does get merged into merged.md. But, as the default group's ar.conf file doesn't contain out custom ones, the active response doesn't work :(

Error message in agent `ossec.log` when `ar.conf` doesn't contain our custom active response name:
```
2020/06/22 16:53:25 ossec-execd: ERROR: (1311): Invalid command name 'ar-linux-iptables-ip-block1800' provided.
````

I hope that makes sense, please do let me know if I'm just being a moron or if you need more information.

Many thanks

**Comment by TomasTurina** [comment]
Hi @32bitbradley,

Every group is located in the manager side inside the `/var/ossec/etc/shared/` directory. By default, all the agents belong to the `default` group if they aren't assigned to another one.

Each group, including the `default` one, has its own files. Some of these files are the following:

- `agent.conf`
- Rootcheck policy files
- SCA policy files

In addition, there is another file that doesn't belong to a specific group. This file is the same for all the groups and is located at `/var/ossec/etc/shared/ar.conf`. The content of the `ar.conf` file depends on the `active-response` configuration of the manager's `ossec.conf` file.

All the files inside a group folder, alongside with the `ar.conf` common file, will be merged together inside the `merged.mg` file, which will be different for each group. This file will be shared with all the agents of the group.

Each agent will unmerge the `merged.mg` file inside the following folder:

- Unix agents: `/var/ossec/etc/shared/`.
- Windows agents: `C:\Program Files (x86)\ossec-agent\shared\`.

All the group files, including the `ar.conf` file, should be present in the `shared` folder of each agent. Then, each agent will look for these files in that folder.

Having said all of this, I can give you two answers to your questions:
- It is not possible to configure custom `ar.conf` files for different groups. This file is created only once and it belongs to all the groups.
- It is correct that the path of the `ar.conf` file at the manager is the same at the Linux agent's. This is because the `merged.mg`, which contains this file, is decompressed at the same folder.

I hope this information helps you clarify this behavior.
